src/lib/tooltip/tooltip.ts

```diff
@@ -392,6 +426,113 @@ export class MdTooltip implements OnDestroy {
       this._tooltipInstance._markForCheck();
     }
   }
+
+
+  /**
+   * Registers the tooltip message for accessibility. If this message has not yet been
+   * registered, a new element will be created in a visually hidden container with the message
+   * as its content. This provides screenreaders a reference the tooltip trigger's aria-describedby.
+   * If an element has already been created for this message, increase that registered message's
+   * reference count.
+   */
+  private _registerA11yMessage() {
+    if (!this.message) { return; }
+
+    const a11yMessage = a11yMessages.get(this.message);
+    if (a11yMessage) {
+      a11yMessage.count++;
+    } else {
+      this._createA11yMessageElement(this.message);
+    }
+
+    this._setAriaDescribedBy();
+  }
+
+  /**
+   * Removes the a11y tooltip message element if no other tooltips are registered with the same
+   * message. Otherwise, decrease the reference count of the registered a11y tooltip message.
+   */
+  private _unregisterA11yMessage(message: string) {
+    if (!this._platform.isBrowser || !this.message) { return; }
+
+    // Remove the a11y message if this was its last unique instance.
+    const a11yMessageElement = a11yMessages.get(message);
+    if (a11yMessageElement && --a11yMessageElement.count == 0) {
```

Comment:
This might be a little more readable as `if (a11yMessageElement && a11yMessageElement.count === 1)` and then doing the decrement separately.